In [1]:
import pandas as pd
df  = pd.read_csv("data/HealthCare  Mini- Project/kidney_disease.csv")

In [2]:
df

id   age    bp     sg   al   su     rbc        pc         pcc  \
0      0  48.0  80.0  1.020  1.0  0.0     NaN    normal  notpresent   
1      1   7.0  50.0  1.020  4.0  0.0     NaN    normal  notpresent   
2      2  62.0  80.0  1.010  2.0  3.0  normal    normal  notpresent   
3      3  48.0  70.0  1.005  4.0  0.0  normal  abnormal     present   
4      4  51.0  80.0  1.010  2.0  0.0  normal    normal  notpresent   
..   ...   ...   ...    ...  ...  ...     ...       ...         ...   
395  395  55.0  80.0  1.020  0.0  0.0  normal    normal  notpresent   
396  396  42.0  70.0  1.025  0.0  0.0  normal    normal  notpresent   
397  397  12.0  80.0  1.020  0.0  0.0  normal    normal  notpresent   
398  398  17.0  60.0  1.025  0.0  0.0  normal    normal  notpresent   
399  399  58.0  80.0  1.025  0.0  0.0  normal    normal  notpresent   

             ba  ...  pcv    wc   rc  htn   dm  cad appet   pe  ane  \
0    notpresent  ...   44  7800  5.2  yes  yes   no  good   no   no   
1    notpresent  ...   38  6000  NaN   no   no   no  good   no   no   
2    notpresent  ...   31  7500  NaN   no  yes   no  poor   no  yes   
3    notpresent  ...   32  6700  3.9  yes   no   no  poor  yes  yes   
4    notpresent  ...   35  7300  4.6   no   no   no  good   no   no   
..          ...  ...  ...   ...  ...  ...  ...  ...   ...  ...  ...   
395  notpresent  ...   47  6700  4.9   no   no   no  good   no   no   
396  notpresent  ...   54  7800  6.2   no   no   no  good   no   no   
397  notpresent  ...   49  6600  5.4   no   no   no  good   no   no   
398  notpresent  ...   51  7200  5.9   no   no   no  good   no   no   
399  notpresent  ...   53  6800  6.1   no   no   no  good   no   no   

    classification  
0              ckd  
1              ckd  
2              ckd  
3              ckd  
4              ckd  
..             ...  
395         notckd  
396         notckd  
397         notckd  
398         notckd  
399         notckd  

[400 rows x 26 columns]

In [3]:
df.describe()

id         age          bp          sg          al          su  \
count  400.000000  391.000000  388.000000  353.000000  354.000000  351.000000   
mean   199.500000   51.483376   76.469072    1.017408    1.016949    0.450142   
std    115.614301   17.169714   13.683637    0.005717    1.352679    1.099191   
min      0.000000    2.000000   50.000000    1.005000    0.000000    0.000000   
25%     99.750000   42.000000   70.000000    1.010000    0.000000    0.000000   
50%    199.500000   55.000000   80.000000    1.020000    0.000000    0.000000   
75%    299.250000   64.500000   80.000000    1.020000    2.000000    0.000000   
max    399.000000   90.000000  180.000000    1.025000    5.000000    5.000000   

              bgr          bu          sc         sod         pot        hemo  
count  356.000000  381.000000  383.000000  313.000000  312.000000  348.000000  
mean   148.036517   57.425722    3.072454  137.528754    4.627244   12.526437  
std     79.281714   50.503006    5.741126   10.408752    3.193904    2.912587  
min     22.000000    1.500000    0.400000    4.500000    2.500000    3.100000  
25%     99.000000   27.000000    0.900000  135.000000    3.800000   10.300000  
50%    121.000000   42.000000    1.300000  138.000000    4.400000   12.650000  
75%    163.000000   66.000000    2.800000  142.000000    4.900000   15.000000  
max    490.000000  391.000000   76.000000  163.000000   47.000000   17.800000

In [4]:
df.columns

Index(['id', 'age', 'bp', 'sg', 'al', 'su', 'rbc', 'pc', 'pcc', 'ba', 'bgr',
       'bu', 'sc', 'sod', 'pot', 'hemo', 'pcv', 'wc', 'rc', 'htn', 'dm', 'cad',
       'appet', 'pe', 'ane', 'classification'],
      dtype='object')

In [5]:
df.isna()

id    age     bp     sg     al     su    rbc     pc    pcc     ba  \
0    False  False  False  False  False  False   True  False  False  False   
1    False  False  False  False  False  False   True  False  False  False   
2    False  False  False  False  False  False  False  False  False  False   
3    False  False  False  False  False  False  False  False  False  False   
4    False  False  False  False  False  False  False  False  False  False   
..     ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
395  False  False  False  False  False  False  False  False  False  False   
396  False  False  False  False  False  False  False  False  False  False   
397  False  False  False  False  False  False  False  False  False  False   
398  False  False  False  False  False  False  False  False  False  False   
399  False  False  False  False  False  False  False  False  False  False   

     ...    pcv     wc     rc    htn     dm    cad  appet     pe    ane  \
0    ...  False  False  False  False  False  False  False  False  False   
1    ...  False  False   True  False  False  False  False  False  False   
2    ...  False  False   True  False  False  False  False  False  False   
3    ...  False  False  False  False  False  False  False  False  False   
4    ...  False  False  False  False  False  False  False  False  False   
..   ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
395  ...  False  False  False  False  False  False  False  False  False   
396  ...  False  False  False  False  False  False  False  False  False   
397  ...  False  False  False  False  False  False  False  False  False   
398  ...  False  False  False  False  False  False  False  False  False   
399  ...  False  False  False  False  False  False  False  False  False   

     classification  
0             False  
1             False  
2             False  
3             False  
4             False  
..              ...  
395           False  
396           False  
397           False  
398           False  
399           False  

[400 rows x 26 columns]

In [6]:
df.isna().sum()

id                  0
age                 9
bp                 12
sg                 47
al                 46
su                 49
rbc               152
pc                 65
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### : Age of the patient in years (e.g. 35, 42, 67, etc.)

In [7]:
df['age'].value_counts()

age
60.0    19
65.0    17
48.0    12
55.0    12
50.0    12
        ..
83.0     1
27.0     1
14.0     1
81.0     1
79.0     1
Name: count, Length: 76, dtype: int64

In [8]:
df['age'].mean()

51.48337595907928

In [9]:
df['age'].median()

55.0

In [10]:
df['age'].fillna(df['age'].mean(),inplace = True)

In [11]:
df['age'].isna().sum()

0

### bp: Blood pressure of the patient in mm/Hg (e.g. 120,80, 140, etc.)

In [12]:
df['bp'].value_counts()

bp
80.0     116
70.0     112
60.0      71
90.0      53
100.0     25
50.0       5
110.0      3
140.0      1
180.0      1
120.0      1
Name: count, dtype: int64

In [13]:
df['bp'].median()

80.0

In [14]:
df['bp'].mean()

76.46907216494846

In [15]:
df['bp'].fillna(df['bp'].median(),inplace = True)

In [16]:
df['bp'].isna().sum()

0

In [17]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                 47
al                 46
su                 49
rbc               152
pc                 65
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### sg: Specific gravity of the patient's urine (e.g. 1.010, 1.020, etc.)

In [18]:
df['sg'].median()

1.02

In [19]:
df['sg'].mean()

1.0174079320113314

In [20]:
#Replacing by Mean
df['sg'].fillna(df['sg'].mean(),inplace =True)

In [21]:
df['sg'].isna().sum()

0

### al: Amount of albumin present in the patient's urine (e.g. trace, 1+, 2+, etc.)

In [22]:
df['al'].value_counts()

al
0.0    199
1.0     44
2.0     43
3.0     43
4.0     24
5.0      1
Name: count, dtype: int64

In [23]:
df['al'].mean()

1.0169491525423728

In [24]:
df['al'].median()

0.0

In [25]:
df['al'].fillna(df['al'].median(), inplace = True)

In [26]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                 49
rbc               152
pc                 65
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### su: Amount of sugar present in the patient's urine (e.g. trace, 1+, 2+, etc.)

In [27]:
df['su'].value_counts()

su
0.0    290
2.0     18
3.0     14
4.0     13
1.0     13
5.0      3
Name: count, dtype: int64

In [28]:
df['su'].mean()

0.45014245014245013

In [29]:
df['su'].median()

0.0

In [30]:
df['su'].fillna(df['su'].median(), inplace = True)

In [31]:
df['su'].isna().sum()

0

### rbc: Presence of red blood cells in the patient's urine (e.g. normal, abnormal)

In [32]:
df['rbc'].value_counts()

rbc
normal      201
abnormal     47
Name: count, dtype: int64

In [33]:
df['rbc'].mode()

0    normal
Name: rbc, dtype: object

In [34]:
df['rbc'].fillna(df['rbc'].mode()[0], inplace = True)

In [35]:
df['rbc'].isna().sum()

0

In [36]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                 65
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### pc: Presence of pus cells in the patient's urine (e.g. normal, abnormal)

In [37]:
df['pc'].value_counts()

pc
normal      259
abnormal     76
Name: count, dtype: int64

In [38]:
df['pc'].mode()

0    normal
Name: pc, dtype: object

In [39]:
df['pc'].fillna(df['pc'].mode()[0], inplace = True)

In [40]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 4
ba                  4
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### pcc: Presence of bacteria in the patient's urine (e.g. present, absent)

In [41]:
df['pcc'].value_counts()

pcc
notpresent    354
present        42
Name: count, dtype: int64

In [42]:
df['pcc'].mode()

0    notpresent
Name: pcc, dtype: object

In [43]:
df['pcc'].fillna(df['pcc'].mode()[0], inplace = True)

In [44]:
df['pcc'].isna().sum()

0

#### Note: Pcc and ba both refer to the presence of bacteria in the patient's urine.
Pcc stands for "pyuria or leukocyte esterase test results", which is a measure of the presence of white blood cells in the urine, and can indicate an infection in the urinary tract
ba: Presence of bacteria in the patient's urine (e.g. present, absent)
Ba stands for "bacteria", and simply indicates whether bacteria were found in the urine or not.
Both pcc and ba are important parameters for diagnosing and monitoring urinary tract infections, which can affect the kidneys and lead to kidney damage if left untreated.

### ba: Presence of bacteria in the patient's urine (e.g. present, absent)

In [45]:
df['ba'].value_counts()

ba
notpresent    374
present        22
Name: count, dtype: int64

In [46]:
df['ba'].fillna(df['ba'].mode()[0], inplace = True)

In [47]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                44
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### bgr: Blood glucose level of the patient in mg/dL (e.g. 120, 200, etc.)

In [48]:
df['bgr'].value_counts()

bgr
99.0     10
93.0      9
100.0     9
107.0     8
131.0     6
         ..
288.0     1
182.0     1
84.0      1
256.0     1
226.0     1
Name: count, Length: 146, dtype: int64

In [49]:
df['bgr'].mean()

148.0365168539326

In [50]:
df['bgr'].median()

121.0

In [51]:
df['bgr'].mode()

0    99.0
Name: bgr, dtype: float64

In [52]:
df['bgr'].fillna(df['bgr'].mode()[0], inplace = True)

In [53]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                 19
sc                 17
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### bu: Blood urea level of the patient in mg/dL (e.g. 20, 30, etc.)

In [54]:
df['bu'].value_counts()

bu
46.0     15
25.0     13
19.0     11
40.0     10
50.0      9
         ..
176.0     1
145.0     1
92.0      1
322.0     1
186.0     1
Name: count, Length: 118, dtype: int64

In [55]:
df['bu'].mode()

0    46.0
Name: bu, dtype: float64

In [56]:
df['bu'].fillna(df['bu'].mode()[0], inplace = True)

In [57]:
df['bu'].isna().sum()

0

### sc: Serum creatinine level of the patient in mg/dL (e.g. 0.8, 1.2, etc.)

In [58]:
df['sc'].value_counts()

sc
1.2     40
1.1     24
0.5     23
1.0     23
0.9     22
        ..
3.8      1
12.2     1
9.2      1
13.8     1
0.4      1
Name: count, Length: 84, dtype: int64

In [59]:
df['sc'].fillna(df['sc'].mode()[0], inplace = True)

In [60]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                87
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### sod: Sodium level in the patient's blood in mEq/L (e.g. 135, 140, etc.)

In [61]:
df['sod'].value_counts()

sod
135.0    40
140.0    25
141.0    22
139.0    21
138.0    20
142.0    20
137.0    19
150.0    17
136.0    17
147.0    13
145.0    11
132.0    10
146.0    10
131.0     9
144.0     9
133.0     8
130.0     7
134.0     6
143.0     4
124.0     3
127.0     3
122.0     2
113.0     2
120.0     2
125.0     2
128.0     2
114.0     2
126.0     1
163.0     1
115.0     1
129.0     1
4.5       1
104.0     1
111.0     1
Name: count, dtype: int64

In [62]:
df['sod'].mode()

0    135.0
Name: sod, dtype: float64

In [63]:
df['sod'].fillna(df['sod'].mode()[0], inplace = True)

In [64]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                88
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### pot: Potassium level in the patient's blood in mEq/L (e.g. 3.5, 4.0, etc.)

In [65]:
df['pot'].value_counts()

pot
3.5     30
5.0     30
4.9     27
4.7     17
4.8     16
3.9     14
3.8     14
4.1     14
4.2     14
4.0     14
4.4     14
4.5     13
4.3     12
3.7     12
3.6      8
4.6      7
3.4      5
5.2      5
5.3      4
5.7      4
3.2      3
5.5      3
6.3      3
5.4      3
2.9      3
3.3      3
5.6      2
3.0      2
6.5      2
2.5      2
5.9      2
5.8      2
7.6      1
47.0     1
6.6      1
5.1      1
6.4      1
2.8      1
2.7      1
39.0     1
Name: count, dtype: int64

In [66]:
df['pot'].mean()

4.62724358974359

In [67]:
df['pot'].median()

4.4

In [68]:
df['pot'].mode()[1]

5.0

In [69]:
df['pot'].fillna(df['pot'].mode()[1], inplace = True)

In [70]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### hemo: Hemoglobin level in the patient's blood in g/dL (e.g. 10, 12, etc.)

In [71]:
df['hemo'].value_counts()

hemo
15.0    16
10.9     8
13.6     7
13.0     7
9.8      7
        ..
6.8      1
8.5      1
7.3      1
12.8     1
17.6     1
Name: count, Length: 115, dtype: int64

In [72]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo               52
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

In [73]:
df['hemo'].fillna(df['hemo'].mean(), inplace=True)

In [74]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo                0
pcv                70
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### pcv: Packed cell volume (PCV) in the patient's blood (e.g. 35, 40, etc.)

In [75]:
df['pcv'].value_counts()

pcv
41      21
52      21
44      19
48      19
40      16
43      14
42      13
45      13
32      12
36      12
33      12
50      12
28      12
34      11
37      11
30       9
29       9
35       9
46       9
31       8
24       7
39       7
26       6
38       5
53       4
51       4
49       4
47       4
54       4
25       3
27       3
22       3
19       2
23       2
15       1
21       1
17       1
20       1
\t43     1
18       1
9        1
14       1
\t?      1
16       1
Name: count, dtype: int64

In [76]:
df['pcv'].dtype

dtype('O')

In [77]:
# Convert 'pcv' column to numeric data type
df['pcv'] = pd.to_numeric(df['pcv'], errors='coerce')
df['pcv'].dtype

dtype('float64')

In [78]:
df['pcv'].isnull().sum()

71

In [79]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo                0
pcv                71
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

In [80]:
df['pcv'].mode()

0    41.0
1    52.0
Name: pcv, dtype: float64

In [81]:
df['pcv'].mean()

38.88449848024316

In [82]:
df['pcv'].fillna(df['pcv'].mode()[0], inplace=True)

In [83]:
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo                0
pcv                 0
wc                105
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### wc: White blood cell (WBC) count in the patient's blood (e.g. 7000, 8000, etc.)

In [84]:
df['wc'].dtype

dtype('O')

In [85]:
df['wc'].value_counts()

wc
9800     11
6700     10
9200      9
9600      9
7200      9
         ..
19100     1
\t?       1
12300     1
14900     1
12700     1
Name: count, Length: 92, dtype: int64

In [86]:
# converting 'wc' column datatype to numeric data type
df['wc'] = pd.to_numeric(df['wc'], errors='coerce')
df['wc'].dtype

dtype('float64')

In [87]:
df['wc'].describe()

count      294.000000
mean      8406.122449
std       2944.474190
min       2200.000000
25%       6500.000000
50%       8000.000000
75%       9800.000000
max      26400.000000
Name: wc, dtype: float64

In [88]:
df['wc'].fillna(df['wc'].mean(), inplace=True)
df.isna().sum()

id                  0
age                 0
bp                  0
sg                  0
al                  0
su                  0
rbc                 0
pc                  0
pcc                 0
ba                  0
bgr                 0
bu                  0
sc                  0
sod                 0
pot                 0
hemo                0
pcv                 0
wc                  0
rc                130
htn                 2
dm                  2
cad                 2
appet               1
pe                  1
ane                 1
classification      0
dtype: int64

### rc: Red blood cell (RBC) count in the patient's blood (e.g. 4.5, 5.0, etc.)

In [89]:
df['rc'].dtype

dtype('O')

In [90]:
df['rc'].unique()

array(['5.2', nan, '3.9', '4.6', '4.4', '5', '4.0', '3.7', '3.8', '3.4',
       '2.6', '2.8', '4.3', '3.2', '3.6', '4', '4.1', '4.9', '2.5', '4.2',
       '4.5', '3.1', '4.7', '3.5', '6.0', '5.0', '2.1', '5.6', '2.3',
       '2.9', '2.7', '8.0', '3.3', '3.0', '3', '2.4', '4.8', '\t?', '5.4',
       '6.1', '6.2', '6.3', '5.1', '5.8', '5.5', '5.3', '6.4', '5.7',
       '5.9', '6.5'], dtype=object)

In [91]:
df['rc'].value_counts()

rc
5.2    18
4.5    16
4.9    14
4.7    11
4.8    10
3.9    10
4.6     9
3.4     9
5.9     8
5.5     8
6.1     8
5.0     8
3.7     8
5.3     7
5.8     7
5.4     7
3.8     7
5.6     6
4.3     6
4.2     6
3.2     5
4.4     5
5.7     5
6.4     5
5.1     5
6.2     5
6.5     5
4.1     5
3.6     4
6.3     4
6.0     4
4.0     3
3.3     3
4       3
3.5     3
2.9     2
3.1     2
2.6     2
2.1     2
2.5     2
2.8     2
3.0     2
2.7     2
5       2
2.3     1
\t?     1
2.4     1
3       1
8.0     1
Name: count, dtype: int64

In [92]:
# changing 'rc' column  to numeric data type
df['rc'] = pd.to_numeric(df['rc'], errors= 'coerce')
df['rc'].dtype

dtype('float64')

In [93]:
df['rc'].fillna(df['rc'].mean(), inplace=True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               2
dm                2
cad               2
appet             1
pe                1
ane               1
classification    0
dtype: int64

### htn: Presence of hypertension (e.g. yes, no)

In [94]:
df['htn'].dtype

dtype('O')

In [95]:
df['htn'].unique()

array(['yes', 'no', nan], dtype=object)

In [96]:
df['htn'].value_counts()

htn
no     251
yes    147
Name: count, dtype: int64

In [97]:
df['htn'].fillna(df['htn'].mode()[0] ,inplace=True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                2
cad               2
appet             1
pe                1
ane               1
classification    0
dtype: int64

### dm: Presence of diabetes mellitus (e.g. yes, no)

In [98]:
df['dm'].unique()

array(['yes', 'no', ' yes', '\tno', '\tyes', nan], dtype=object)

In [99]:
df['dm'].value_counts()

dm
no       258
yes      134
\tno       3
\tyes      2
 yes       1
Name: count, dtype: int64

In [100]:
df['dm'] = df['dm'].replace('\tno' , 'no')
df['dm'] = df['dm'].replace(['\tyes',' yes'] , 'yes')
df['dm'].value_counts()

dm
no     261
yes    137
Name: count, dtype: int64

In [101]:
df['dm'].isna().sum()

2

In [102]:
df['dm'].fillna(df['dm'].mode()[0], inplace = True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                0
cad               2
appet             1
pe                1
ane               1
classification    0
dtype: int64

### cad: Presence of coronary artery disease (e.g. yes, no)

In [103]:
df['cad'].value_counts()

cad
no      362
yes      34
\tno      2
Name: count, dtype: int64

In [104]:
df['cad'] = df['cad'].replace('\tno', 'no')

In [105]:
df['cad'].value_counts()

cad
no     364
yes     34
Name: count, dtype: int64

In [106]:
df['cad'].mode()[0]

'no'

In [107]:
df['cad'].fillna(df['cad'].mode()[0], inplace = True)

In [108]:
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                0
cad               0
appet             1
pe                1
ane               1
classification    0
dtype: int64

### appet: Appetite of the patient (e.g. good, poor)

In [109]:
df['appet'].dtype

dtype('O')

In [110]:
df['appet'].unique()

array(['good', 'poor', nan], dtype=object)

In [111]:
df['appet'].value_counts()

appet
good    317
poor     82
Name: count, dtype: int64

In [112]:
df['appet'].fillna(df['appet'].mode()[0], inplace=True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                0
cad               0
appet             0
pe                1
ane               1
classification    0
dtype: int64

### pe: Presence of pedal edema (e.g. yes, no)

In [113]:
df['pe'].unique()

array(['no', 'yes', nan], dtype=object)

In [114]:
df['pe'].value_counts()

pe
no     323
yes     76
Name: count, dtype: int64

In [115]:
df['pe'].fillna(df['pe'].mode()[0], inplace=True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                0
cad               0
appet             0
pe                0
ane               1
classification    0
dtype: int64

### ane: Presence of anemia (e.g. yes, no)

In [116]:
df['ane'].unique()

array(['no', 'yes', nan], dtype=object)

In [117]:
df['ane'].fillna(df['ane'].mode()[0], inplace=True)
df.isna().sum()

id                0
age               0
bp                0
sg                0
al                0
su                0
rbc               0
pc                0
pcc               0
ba                0
bgr               0
bu                0
sc                0
sod               0
pot               0
hemo              0
pcv               0
wc                0
rc                0
htn               0
dm                0
cad               0
appet             0
pe                0
ane               0
classification    0
dtype: int64

### classification: Diagnosis of kidney disease (e.g. ckd as chronic kidney disease, notckd as not chronic kidney disease , etc.)

In [118]:
df['classification'].unique()

array(['ckd', 'ckd\t', 'notckd'], dtype=object)

In [119]:
df['classification'].value_counts()

classification
ckd       248
notckd    150
ckd\t       2
Name: count, dtype: int64

In [120]:
df['classification'] = df['classification'].replace('ckd\t', 'ckd')

In [121]:
df['classification'].unique()

array(['ckd', 'notckd'], dtype=object)

In [122]:
df['classification'].value_counts()

classification
ckd       250
notckd    150
Name: count, dtype: int64

In [123]:
df.to_csv('CleanedKidneyData.csv')

# Machine Learning Model (Classification)

In [124]:
df.describe().columns

Index(['id', 'age', 'bp', 'sg', 'al', 'su', 'bgr', 'bu', 'sc', 'sod', 'pot',
       'hemo', 'pcv', 'wc', 'rc'],
      dtype='object')

In [125]:
# df1 is a dataframe consisting of only categorical datatypes
df1 = df.drop(df.describe().columns, axis = 1)

In [126]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   rbc             400 non-null    object
 1   pc              400 non-null    object
 2   pcc             400 non-null    object
 3   ba              400 non-null    object
 4   htn             400 non-null    object
 5   dm              400 non-null    object
 6   cad             400 non-null    object
 7   appet           400 non-null    object
 8   pe              400 non-null    object
 9   ane             400 non-null    object
 10  classification  400 non-null    object
dtypes: object(11)
memory usage: 34.5+ KB


In [127]:
for column in df1.columns:
    print(df1[column].unique())

['normal' 'abnormal']
['normal' 'abnormal']
['notpresent' 'present']
['notpresent' 'present']
['yes' 'no']
['yes' 'no']
['no' 'yes']
['good' 'poor']
['no' 'yes']
['no' 'yes']
['ckd' 'notckd']


In [128]:
df

id   age    bp     sg   al   su     rbc        pc         pcc  \
0      0  48.0  80.0  1.020  1.0  0.0  normal    normal  notpresent   
1      1   7.0  50.0  1.020  4.0  0.0  normal    normal  notpresent   
2      2  62.0  80.0  1.010  2.0  3.0  normal    normal  notpresent   
3      3  48.0  70.0  1.005  4.0  0.0  normal  abnormal     present   
4      4  51.0  80.0  1.010  2.0  0.0  normal    normal  notpresent   
..   ...   ...   ...    ...  ...  ...     ...       ...         ...   
395  395  55.0  80.0  1.020  0.0  0.0  normal    normal  notpresent   
396  396  42.0  70.0  1.025  0.0  0.0  normal    normal  notpresent   
397  397  12.0  80.0  1.020  0.0  0.0  normal    normal  notpresent   
398  398  17.0  60.0  1.025  0.0  0.0  normal    normal  notpresent   
399  399  58.0  80.0  1.025  0.0  0.0  normal    normal  notpresent   

             ba  ...   pcv      wc        rc  htn   dm  cad  appet   pe  ane  \
0    notpresent  ...  44.0  7800.0  5.200000  yes  yes   no   good   no   no   
1    notpresent  ...  38.0  6000.0  4.707435   no   no   no   good   no   no   
2    notpresent  ...  31.0  7500.0  4.707435   no  yes   no   poor   no  yes   
3    notpresent  ...  32.0  6700.0  3.900000  yes   no   no   poor  yes  yes   
4    notpresent  ...  35.0  7300.0  4.600000   no   no   no   good   no   no   
..          ...  ...   ...     ...       ...  ...  ...  ...    ...  ...  ...   
395  notpresent  ...  47.0  6700.0  4.900000   no   no   no   good   no   no   
396  notpresent  ...  54.0  7800.0  6.200000   no   no   no   good   no   no   
397  notpresent  ...  49.0  6600.0  5.400000   no   no   no   good   no   no   
398  notpresent  ...  51.0  7200.0  5.900000   no   no   no   good   no   no   
399  notpresent  ...  53.0  6800.0  6.100000   no   no   no   good   no   no   

    classification  
0              ckd  
1              ckd  
2              ckd  
3              ckd  
4              ckd  
..             ...  
395         notckd  
396         notckd  
397         notckd  
398         notckd  
399         notckd  

[400 rows x 26 columns]

In [129]:
df.columns

Index(['id', 'age', 'bp', 'sg', 'al', 'su', 'rbc', 'pc', 'pcc', 'ba', 'bgr',
       'bu', 'sc', 'sod', 'pot', 'hemo', 'pcv', 'wc', 'rc', 'htn', 'dm', 'cad',
       'appet', 'pe', 'ane', 'classification'],
      dtype='object')

In [130]:
#Splitting the Data
X = df.drop(['id','age','classification'],axis=1)
y = df['classification']

In [131]:
X

bp     sg   al   su     rbc        pc         pcc          ba    bgr  \
0    80.0  1.020  1.0  0.0  normal    normal  notpresent  notpresent  121.0   
1    50.0  1.020  4.0  0.0  normal    normal  notpresent  notpresent   99.0   
2    80.0  1.010  2.0  3.0  normal    normal  notpresent  notpresent  423.0   
3    70.0  1.005  4.0  0.0  normal  abnormal     present  notpresent  117.0   
4    80.0  1.010  2.0  0.0  normal    normal  notpresent  notpresent  106.0   
..    ...    ...  ...  ...     ...       ...         ...         ...    ...   
395  80.0  1.020  0.0  0.0  normal    normal  notpresent  notpresent  140.0   
396  70.0  1.025  0.0  0.0  normal    normal  notpresent  notpresent   75.0   
397  80.0  1.020  0.0  0.0  normal    normal  notpresent  notpresent  100.0   
398  60.0  1.025  0.0  0.0  normal    normal  notpresent  notpresent  114.0   
399  80.0  1.025  0.0  0.0  normal    normal  notpresent  notpresent  131.0   

       bu  ...  hemo   pcv      wc        rc  htn   dm  cad appet   pe  ane  
0    36.0  ...  15.4  44.0  7800.0  5.200000  yes  yes   no  good   no   no  
1    18.0  ...  11.3  38.0  6000.0  4.707435   no   no   no  good   no   no  
2    53.0  ...   9.6  31.0  7500.0  4.707435   no  yes   no  poor   no  yes  
3    56.0  ...  11.2  32.0  6700.0  3.900000  yes   no   no  poor  yes  yes  
4    26.0  ...  11.6  35.0  7300.0  4.600000   no   no   no  good   no   no  
..    ...  ...   ...   ...     ...       ...  ...  ...  ...   ...  ...  ...  
395  49.0  ...  15.7  47.0  6700.0  4.900000   no   no   no  good   no   no  
396  31.0  ...  16.5  54.0  7800.0  6.200000   no   no   no  good   no   no  
397  26.0  ...  15.8  49.0  6600.0  5.400000   no   no   no  good   no   no  
398  50.0  ...  14.2  51.0  7200.0  5.900000   no   no   no  good   no   no  
399  18.0  ...  15.8  53.0  6800.0  6.100000   no   no   no  good   no   no  

[400 rows x 23 columns]

In [132]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test= train_test_split(X,y, test_size=0.2)

In [133]:
X_train

bp        sg   al   su       rbc        pc         pcc          ba  \
285   70.0  1.020000  0.0  0.0    normal    normal  notpresent  notpresent   
256   80.0  1.025000  0.0  0.0    normal    normal  notpresent  notpresent   
237   70.0  1.015000  2.0  2.0    normal    normal  notpresent  notpresent   
101   90.0  1.015000  2.0  0.0    normal  abnormal     present     present   
76    80.0  1.005000  4.0  0.0  abnormal  abnormal  notpresent     present   
..     ...       ...  ...  ...       ...       ...         ...         ...   
73   100.0  1.015000  2.0  0.0  abnormal  abnormal  notpresent  notpresent   
122   70.0  1.017408  0.0  0.0    normal    normal  notpresent  notpresent   
328   70.0  1.020000  0.0  0.0    normal    normal  notpresent  notpresent   
209   70.0  1.020000  0.0  0.0    normal    normal  notpresent  notpresent   
350   70.0  1.025000  0.0  0.0    normal    normal  notpresent  notpresent   

       bgr     bu  ...       hemo   pcv            wc        rc  htn   dm  \
285   94.0   19.0  ...  16.000000  41.0   5300.000000  5.900000   no   no   
256  131.0   10.0  ...  14.500000  41.0  10700.000000  5.100000   no   no   
237  141.0   53.0  ...  12.700000  40.0   9600.000000  4.707435  yes  yes   
101   88.0   80.0  ...  11.300000  33.0  10700.000000  3.900000   no   no   
76   133.0  139.0  ...  10.300000  36.0   6200.000000  4.000000   no  yes   
..     ...    ...  ...        ...   ...           ...       ...  ...  ...   
73   129.0  107.0  ...   4.800000  14.0   6300.000000  4.707435  yes   no   
122   99.0  219.0  ...   6.000000  41.0   8406.122449  4.707435  yes   no   
328  131.0   29.0  ...  12.526437  45.0   8600.000000  6.500000   no   no   
209   99.0   46.0  ...  11.500000  41.0   6900.000000  4.707435   no   no   
350   85.0   20.0  ...  16.100000  43.0   9600.000000  4.500000   no   no   

     cad appet   pe  ane  
285   no  good   no   no  
256   no  good   no   no  
237   no  poor  yes   no  
101   no  good   no   no  
76    no  good  yes   no  
..   ...   ...  ...  ...  
73    no  good  yes  yes  
122   no  good   no  yes  
328   no  good   no   no  
209   no  good   no   no  
350   no  good   no   no  

[320 rows x 23 columns]

In [134]:
y_train

285    notckd
256    notckd
237       ckd
101       ckd
76        ckd
        ...  
73        ckd
122       ckd
328    notckd
209       ckd
350    notckd
Name: classification, Length: 320, dtype: object

In [135]:
X_test

bp     sg   al   su       rbc        pc         pcc          ba  \
307   60.0  1.020  0.0  0.0    normal    normal  notpresent  notpresent   
246  110.0  1.015  3.0  0.0  abnormal    normal     present  notpresent   
335   60.0  1.020  0.0  0.0    normal    normal  notpresent  notpresent   
287   70.0  1.025  0.0  0.0    normal    normal  notpresent  notpresent   
87   100.0  1.005  1.0  0.0    normal  abnormal     present  notpresent   
..     ...    ...  ...  ...       ...       ...         ...         ...   
224   60.0  1.020  0.0  0.0    normal    normal  notpresent  notpresent   
397   80.0  1.020  0.0  0.0    normal    normal  notpresent  notpresent   
7     80.0  1.015  2.0  4.0    normal  abnormal  notpresent  notpresent   
377   70.0  1.020  0.0  0.0    normal    normal  notpresent  notpresent   
77    70.0  1.010  1.0  0.0    normal    normal  notpresent  notpresent   

       bgr     bu  ...       hemo   pcv            wc        rc  htn   dm  \
307  137.0   17.0  ...  13.600000  44.0   7900.000000  4.500000   no   no   
246  106.0  215.0  ...   8.600000  26.0   5000.000000  2.500000  yes   no   
335  134.0   45.0  ...  14.200000  48.0  10700.000000  5.600000   no   no   
287  124.0   22.0  ...  13.400000  43.0   8406.122449  4.707435   no   no   
87   169.0   47.0  ...  11.100000  32.0   5800.000000  5.000000  yes  yes   
..     ...    ...  ...        ...   ...           ...       ...  ...  ...   
224  117.0   28.0  ...  12.526437  41.0   8406.122449  4.707435   no   no   
397  100.0   26.0  ...  15.800000  49.0   6600.000000  5.400000   no   no   
7    410.0   31.0  ...  12.400000  44.0   6900.000000  5.000000   no  yes   
377   97.0   27.0  ...  13.800000  49.0   6400.000000  4.800000   no   no   
77   102.0   48.0  ...  11.900000  34.0   7100.000000  3.700000  yes  yes   

     cad appet   pe  ane  
307   no  good   no   no  
246  yes  good   no  yes  
335   no  good   no   no  
287   no  good   no   no  
87    no  poor   no   no  
..   ...   ...  ...  ...  
224   no  good  yes   no  
397   no  good   no   no  
7     no  good  yes   no  
377   no  good   no   no  
77    no  good  yes   no  

[80 rows x 23 columns]

In [136]:
y_test

307    notckd
246       ckd
335    notckd
287    notckd
87        ckd
        ...  
224       ckd
397    notckd
7         ckd
377    notckd
77        ckd
Name: classification, Length: 80, dtype: object

In [137]:
print('Shape of X_train: ', X_train.shape)
print('Shape of X_test: ', X_test.shape)
print('Shape of y_train: ',y_train.shape)
print('Shape of y_test : ', y_test.shape)

Shape of X_train:  (320, 23)
Shape of X_test:  (80, 23)
Shape of y_train:  (320,)
Shape of y_test :  (80,)


#### Refered and Modified Little

In [166]:
# Encode the categorical features using OneHotEncoder
categorical_cols = ['rbc', 'pc', 'pcc', 'ba', 'htn', 'dm', 'cad', 'pe', 'ane']
from sklearn.preprocessing import OneHotEncoder
onehot_encoder = OneHotEncoder(drop='first', sparse=False)
# fit_transform on X_train only
onehot_encoder.fit(X_train[categorical_cols])
cols = onehot_encoder.get_feature_names_out(categorical_cols) #just for Example, cols contain new columns  name after one hot Encoding, Can be passed as a argument while transform
# Transform on X_train
X_train_cat_encoded = pd.DataFrame(onehot_encoder.transform(X_train[categorical_cols]),columns=categorical_cols)  #Creating a Data frame for Encoded Categoricval Values
# Transform on X_test
X_test_cat_encoded = pd.DataFrame(onehot_encoder.transform(X_test[categorical_cols]), columns=categorical_cols)   #Creating a Data frame for Encoded Categoricval Values

C:\Users\91877\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\preprocessing\_encoders.py:975: FutureWarning: `sparse` was renamed to `sparse_output` in version 1.2 and will be removed in 1.4. `sparse_output` is ignored unless you leave `sparse` to its default value.
  warnings.warn(


In [170]:
X_train[categorical_cols]

rbc        pc         pcc          ba  htn   dm cad   pe  ane
285    normal    normal  notpresent  notpresent   no   no  no   no   no
256    normal    normal  notpresent  notpresent   no   no  no   no   no
237    normal    normal  notpresent  notpresent  yes  yes  no  yes   no
101    normal  abnormal     present     present   no   no  no   no   no
76   abnormal  abnormal  notpresent     present   no  yes  no  yes   no
..        ...       ...         ...         ...  ...  ...  ..  ...  ...
73   abnormal  abnormal  notpresent  notpresent  yes   no  no  yes  yes
122    normal    normal  notpresent  notpresent  yes   no  no   no  yes
328    normal    normal  notpresent  notpresent   no   no  no   no   no
209    normal    normal  notpresent  notpresent   no   no  no   no   no
350    normal    normal  notpresent  notpresent   no   no  no   no   no

[320 rows x 9 columns]

In [171]:
cols #Can be passed as a argument while one hot encoder transform to change name of columns as cols

array(['rbc_normal', 'pc_normal', 'pcc_present', 'ba_present', 'htn_yes',
       'dm_yes', 'cad_yes', 'pe_yes', 'ane_yes'], dtype=object)

In [172]:
X_train_cat_encoded

rbc   pc  pcc   ba  htn   dm  cad   pe  ane
0    1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
1    1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
2    1.0  1.0  0.0  0.0  1.0  1.0  0.0  1.0  0.0
3    1.0  0.0  1.0  1.0  0.0  0.0  0.0  0.0  0.0
4    0.0  0.0  0.0  1.0  0.0  1.0  0.0  1.0  0.0
..   ...  ...  ...  ...  ...  ...  ...  ...  ...
315  0.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  1.0
316  1.0  1.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
317  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
318  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
319  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0

[320 rows x 9 columns]

In [184]:
X_test_cat_encoded

rbc   pc  pcc   ba  htn   dm  cad   pe  ane
0   1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
1   0.0  1.0  1.0  0.0  1.0  0.0  1.0  0.0  1.0
2   1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
3   1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
4   1.0  0.0  1.0  0.0  1.0  1.0  0.0  0.0  0.0
..  ...  ...  ...  ...  ...  ...  ...  ...  ...
75  1.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0  0.0
76  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
77  1.0  0.0  0.0  0.0  0.0  1.0  0.0  1.0  0.0
78  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
79  1.0  1.0  0.0  0.0  1.0  1.0  0.0  1.0  0.0

[80 rows x 9 columns]

In [185]:
# Encode the ordinal feature using OrdinalEncoder, Same Steps as one hot  Encoder
ordinal_cols = ['appet']
ordinal_categories = [['poor', 'good']]
from sklearn.preprocessing import OrdinalEncoder 
ordinal_encoder = OrdinalEncoder(categories=ordinal_categories)
# transform on X_train
X_train_ord_encoded = pd.DataFrame(ordinal_encoder.fit_transform(X_train[ordinal_cols]), columns=ordinal_cols)
# only transform on X_test
X_test_ord_encoded = pd.DataFrame(ordinal_encoder.transform(X_test[ordinal_cols]), columns=ordinal_cols)

In [187]:

# Combine the encoded features into a single DataFrame
# X_train
X_train_encoded = pd.concat([X_train.drop(columns=categorical_cols+ordinal_cols).reset_index(),
                             X_train_cat_encoded,X_train_ord_encoded],axis=1).drop('index',axis=1)
# X_test
X_test_encoded = pd.concat([X_test.drop(columns=categorical_cols+ordinal_cols).reset_index(),
                            X_test_cat_encoded, X_test_ord_encoded], axis=1).drop('index',axis=1)

In [218]:
from sklearn.preprocessing import LabelEncoder
label_encoder = LabelEncoder()
label_encoder.fit(y_train)
y_train_encoded = label_encoder.transform(y_train)
y_test_encoded = label_encoder.transform(y_test)

In [220]:
print(label_encoder.classes_)  #ckd = 0 and notckd = 1

['ckd' 'notckd']


In [221]:
from sklearn.preprocessing import StandardScaler
std_scaler = StandardScaler()
std_scaler.fit(X_train_encoded)

X_train_scaled = pd.DataFrame(std_scaler.transform(X_train_encoded),
                              columns=X_train_encoded.columns)

X_test_scaled = pd.DataFrame(std_scaler.transform(X_test_encoded),
                             columns= X_test_encoded.columns)

In [222]:
X_train_scaled

bp        sg        al        su       bgr        bu        sc  \
0   -0.501755  0.476397 -0.694170 -0.368421 -0.609200 -0.789120 -0.498812   
1    0.293105  1.413381 -0.694170 -0.368421 -0.112920 -0.977360 -0.545780   
2   -0.501755 -0.460587  0.806739  1.736842  0.021209 -0.077989 -0.146554   
3    1.087965 -0.460587  0.806739 -0.368421 -0.689678  0.486732  0.370091   
4    0.293105 -2.334555  2.307648 -0.368421 -0.086094  1.720753  1.332929   
..        ...       ...       ...       ...       ...       ...       ...   
315  1.882825 -0.460587  0.806739 -0.368421 -0.139746  1.051454  0.910220   
316 -0.501755 -0.009348 -0.694170 -0.368421 -0.542135  3.394002  2.201832   
317 -0.501755  0.476397 -0.694170 -0.368421 -0.112920 -0.579964 -0.522296   
318 -0.501755  0.476397 -0.694170 -0.368421 -0.542135 -0.224398 -0.381393   
319 -0.501755  1.413381 -0.694170 -0.368421 -0.729917 -0.768204 -0.428360   

          sod       pot      hemo  ...       rbc        pc       pcc  \
0   -0.339041 -0.299359  1.274262  ...  0.372545  0.490214 -0.339091   
1    1.378115  0.144696  0.729879  ...  0.372545  0.490214 -0.339091   
2   -0.339041  0.144696  0.076619  ...  0.372545  0.490214 -0.339091   
3    0.285380  0.427277 -0.431472  ...  0.372545 -2.039924  2.949063   
4   -0.807356  0.346540 -0.794394  ... -2.684237 -2.039924 -0.339091   
..        ...       ...       ...  ...       ...       ...       ...   
315 -0.807356 -0.097516 -2.790466  ... -2.684237 -2.039924 -0.339091   
316 -1.119566 -0.339728 -2.354960  ...  0.372545  0.490214 -0.339091   
317  1.222010  0.104328  0.013629  ...  0.372545  0.490214 -0.339091   
318 -0.339041  0.144696 -0.358888  ...  0.372545  0.490214 -0.339091   
319  0.753695  0.063959  1.310555  ...  0.372545  0.490214 -0.339091   

           ba       htn        dm       cad        pe       ane     appet  
0   -0.265017 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
1   -0.265017 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
2   -0.265017  1.344254  1.430950 -0.297381  2.060559 -0.430331 -1.943224  
3    3.773340 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
4    3.773340 -0.743907  1.430950 -0.297381  2.060559 -0.430331  0.514609  
..        ...       ...       ...       ...       ...       ...       ...  
315 -0.265017  1.344254 -0.698836 -0.297381  2.060559  2.323790  0.514609  
316 -0.265017  1.344254 -0.698836 -0.297381 -0.485305  2.323790  0.514609  
317 -0.265017 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
318 -0.265017 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
319 -0.265017 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  

[320 rows x 23 columns]

In [223]:
X_test_scaled

bp        sg        al        su       bgr        bu        sc  \
0  -1.296615  0.476397 -0.694170 -0.368421 -0.032443 -0.830951 -0.545780   
1   2.677685 -0.460587  1.557193 -0.368421 -0.448244  3.310340  2.906348   
2  -1.296615  0.476397 -0.694170 -0.368421 -0.072681 -0.245314 -0.545780   
3  -0.501755  1.413381 -0.694170 -0.368421 -0.206811 -0.726373 -0.522296   
4   1.882825 -2.334555  0.056284 -0.368421  0.396772 -0.203483  0.017833   
..       ...       ...       ...       ...       ...       ...       ...   
75 -1.296615  0.476397 -0.694170 -0.368421 -0.300702 -0.600879 -0.146554   
76  0.293105  0.476397 -0.694170 -0.368421 -0.528722 -0.642711 -0.522296   
77  0.293105 -0.460587  0.806739  3.842105  3.629295 -0.538132 -0.404876   
78 -0.501755  0.476397 -0.694170 -0.368421 -0.568961 -0.621795 -0.498812   
79 -0.501755 -1.397571  0.056284 -0.368421 -0.501896 -0.182567  0.088285   

         sod       pot      hemo  ...       rbc        pc       pcc        ba  \
0   2.002536 -0.460834  0.403249  ...  0.372545  0.490214 -0.339091 -0.265017   
1  -2.680617  0.427277 -1.411362  ... -2.684237  0.490214  2.949063 -0.265017   
2   0.285380  0.063959  0.621002  ...  0.372545  0.490214 -0.339091 -0.265017   
3  -0.026831 -0.339728  0.330665  ...  0.372545  0.490214 -0.339091 -0.265017   
4  -0.339041  0.144696 -0.504056  ...  0.372545 -2.039924  2.949063 -0.265017   
..       ...       ...       ...  ...       ...       ...       ...       ...   
75  0.129275 -0.339728  0.013629  ...  0.372545  0.490214 -0.339091 -0.265017   
76 -0.026831 -0.097516  1.201678  ...  0.372545  0.490214 -0.339091 -0.265017   
77 -0.339041  0.144696 -0.032258  ...  0.372545 -2.039924 -0.339091 -0.265017   
78  1.222010  0.063959  0.475834  ...  0.372545  0.490214 -0.339091 -0.265017   
79 -0.026831  0.144696 -0.213719  ...  0.372545  0.490214 -0.339091 -0.265017   

         htn        dm       cad        pe       ane     appet  
0  -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
1   1.344254 -0.698836  3.362691 -0.485305  2.323790  0.514609  
2  -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
3  -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
4   1.344254  1.430950 -0.297381 -0.485305 -0.430331 -1.943224  
..       ...       ...       ...       ...       ...       ...  
75 -0.743907 -0.698836 -0.297381  2.060559 -0.430331  0.514609  
76 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
77 -0.743907  1.430950 -0.297381  2.060559 -0.430331  0.514609  
78 -0.743907 -0.698836 -0.297381 -0.485305 -0.430331  0.514609  
79  1.344254  1.430950 -0.297381  2.060559 -0.430331  0.514609  

[80 rows x 23 columns]

In [224]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, accuracy_score
# Logistic on standard scaled 
lg_std = LogisticRegression()
lg_std.fit(X_train_scaled, y_train_encoded) #Fitting Model
y_pred_lg_std  = lg_std.predict(X_test_scaled) #Predicting
lg_std_cm = confusion_matrix(y_test_encoded, y_pred_lg_std)
lg_std_accuracy = accuracy_score(y_test_encoded, y_pred_lg_std)

In [225]:
# for standard sclaed data
print('Logistic Regression with Standard scaler: ')
print(lg_std_cm)
print('accuracy score for standard scaled logistic regression : ', lg_std_accuracy)
print('--------')  

Logistic Regression with Standard scaler: 
[[47  0]
 [ 0 33]]
accuracy score for standard scaled logistic regression :  1.0
--------


In [226]:
lg_std.predict(X_test_scaled) #Predicting, 0= ckd, 1 = notckd

array([1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0])

In [228]:
y_test_encoded

array([1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0])